src/main/kotlin/org/rust/ide/refactoring/move/RsMoveTopLevelItemsHandler.kt

```diff
@@ -84,3 +85,28 @@ class RsMoveTopLevelItemsHandler : MoveHandlerDelegate() {
         }
     }
 }
+
+private fun collectRelatedImplItems(containingMod: RsMod, items: List<RsItemElement>): List<RsImplItem> {
+    // For struct `Foo` we should collect:
+    // * impl Foo { ... }
+    // * impl ... for Foo { ... }
+    // * Maybe also `impl From<Foo> for Bar { ... }`?
+    //   if `Bar` belongs to same crate (but to different module from `Foo`)
+    //
+    // For trait `Foo` we should collect:
+    // * impl Foo for ... { ... }
+    return groupImplsByStructOrTrait(containingMod, items).values.flatten()
+}
+
+private fun groupImplsByStructOrTrait(containingMod: RsMod, items: List<RsItemElement>): Map<RsItemElement, List<RsImplItem>> {
```

Comment:
[minor] Probably  `items: Set<RsItemElement>` is preferable as we only do lookups